Planner/src/Planner.java

```diff
@@ -114,15 +173,92 @@ public void actionPerformed(ActionEvent arg0) {
 		plannerFrame.add(listPanel, BorderLayout.LINE_START);
+		
+		//show a dialog when we click on the user in User's list
+		listWithUsers.addListSelectionListener(new ListSelectionListener() {
+			public void valueChanged(ListSelectionEvent evtU) {
+				JTextField userName = new JTextField(25);
+			    JTextField userSurname = new JTextField(25);
+			    JPanel userPanel = new JPanel();
+			    userPanel.add(new JLabel("Edit name:"));
+			    userPanel.add(userName);
+			    userPanel.add(Box.createVerticalStrut(15)); // a spacer
+			    userPanel.add(new JLabel("Edit surname:"));
+			    userPanel.add(userSurname);
+			    userPanel.setLayout(new BoxLayout(userPanel, BoxLayout.Y_AXIS));
+				if (evtU.getValueIsAdjusting() == false) {
+					 int result = JOptionPane.showConfirmDialog(null, userPanel, 
+				               "Edit the User", JOptionPane.OK_CANCEL_OPTION);
+					 if (result == JOptionPane.OK_OPTION) {
+				         System.out.println("Name: " + userName.getText());
+				         System.out.println("Surname: " + userSurname.getText());
+				         User editUser = (User)listWithUsers.getSelectedValue();
+				         editUser.setName(userName.getText());
+				         editUser.setSurname(userSurname.getText());
+					 }
+				}
+			}
+		});
+		
+		//show a dialog when we click on the task
+		listWithTasks.addListSelectionListener(new ListSelectionListener() {
+			public void valueChanged(ListSelectionEvent evtT) {
+				JComboBox selectTask = new JComboBox(tasksList);
+				JComboBox assignedTo = new JComboBox(usersList);
+				JTextField start = new JTextField(30);
+			    JTextField end = new JTextField(30);
+				JTextField taskName = new JTextField(25);
+			    JTextField taskDescription = new JTextField(30);
+				JPanel taskPanel = new JPanel();
+			    taskPanel.add(new JLabel("Edit name of task:"));
+			    taskPanel.add(taskName);
+			    taskPanel.add(Box.createVerticalStrut(15)); // a spacer
+			    taskPanel.add(new JLabel("Edit description of task:"));
+			    taskPanel.add(taskDescription);
+			    taskPanel.add(new JLabel("Choose the user:"));
+			    taskPanel.add(assignedTo);
+			    taskPanel.add(new JLabel("Edit start time. Enter dd/mm/yyyy hh:mm")); 
+			    taskPanel.add(start); // change to field where user can select the date and time
+			    taskPanel.add(new JLabel("Edit end time. Enter dd/mm/yyyy hh:mm"));
+			    taskPanel.add(end); // change to field where user can select the date and time
+			    taskPanel.setLayout(new BoxLayout(taskPanel, BoxLayout.Y_AXIS));
+				if (evtT.getValueIsAdjusting() == false) {
+					 int result = JOptionPane.showConfirmDialog(null, taskPanel, 
+				               "Edit the task", JOptionPane.OK_CANCEL_OPTION);
+					 if (result == JOptionPane.OK_OPTION) {
+				         Task editTask = (Task)listWithTasks.getSelectedValue();
+				         editTask.setNameTask(taskName.getText());
+				         editTask.setStartTime(start.getText());
+				         editTask.setEndTime(end.getText());
```

Comment:
Here you should also save the new description, so you should add editTask.setDescription(taskDescription.getText()).

Reply:
Also please add a space before all the lines that are inside this if.

Reply:
I had to forgot about task description. And I also have the spaces here.